spring-kafka/src/main/java/org/springframework/kafka/listener/PartitionPausingBackoffManager.java

```diff
@@ -229,4 +219,11 @@ protected void removeBackoff(TopicPartition topicPartition) {
 			this.backOffContexts.remove(topicPartition);
 		}
 	}
+
+	@Override
+	public void destroy() throws Exception {
+		if (this.kafkaConsumerTimingAdjuster instanceof DisposableBean) {
+			((DisposableBean) this.kafkaConsumerTimingAdjuster).destroy();
```

Comment:
If we remove `DisposableBean` on the `WakingKafkaConsumerTimingAdjuster`, we don't need this delegation.
Nevertheless the question is still why it is here anyway?
Isn't that  `WakingKafkaConsumerTimingAdjuster` injected into this manager a bean, so its disposition is controlled from the outside anyway?

Reply:
Not really, in the current `Support` implementation the `KafkaBackOffManagerFactory` instantiates it manually, so the container is not aware of it. I think this goes well with the purpose of not registering  beans that are not strictly necessary to be as such, but I can revert it back to a bean approach if that's better.

I don't seem to be able to get a permalink from the PR, but the logic in the factory is:

```java
return this.taskExecutor != null
 		? new WakingKafkaConsumerTimingAdjuster(this.taskExecutor)
 		: new WakingKafkaConsumerTimingAdjuster();
```